### ISSUE TITLE
Add `null` in possible generic types of `object`

### ISSUE THREAD

**Issue by victorgarciaesgi** [dwightjack/vue-types#206]
Hi! 

I'm using `object` for defining a non-required prop. For me a non-required prop could also be of type null. And it's valid using the classic way `PropType` 

```ts
props: {
    selectedCardType: {
         type: Object as PropType<CreditCardType | null> // This works,
    },
    selectedCardType: object<CreditCardType | null>() // Does not work,
}
```

So what I would want to do is adding the possibility of doing `object<CreditCardType | null>()`

It is rather important because using Volar  VScode extension, if you pass a null propertie  to a non-required prop, the types will fail even if this is valid Vue.

I already contributed here, so tell me if you don't have time I can make a PR :)

**Comment by dwightjack** [comment]
Hm, I think that `PropType<CreditCardType | null>` is basically defining a prop that can be either an object or `null`. In this case, I guess the best solution would be to use `oneOfType` 🤔. We just miss a `nullType` definition which could be something like:

```ts
const nullType = () => { type: null as unknown as PropType<null>}
```

Usage in Vue3: 
```ts
<script lang="ts">
import { object, oneOfType, nullType } from 'vue-types'
</script>
<script setup lang="ts">
defineProps({
  abc: oneOfType([object(), nullType()])
})
</script>
```

I think that in this way it would be more explicit. What do you think?

**Comment by victorgarciaesgi** [comment]
Oh yeah that will be even better and more explicit! And it works with all the prop types too :D 

Thanks a lot!

**Comment by dwightjack** [comment]
@victorgarciaesgi I submitted a PR with a new `isNull` validator (didn't come up with a better name) that should fit this scenario: https://github.com/dwightjack/vue-types/pull/215

By the way, I also added you as a collaborator, so feel free to review the PR.

**Comment by victorgarciaesgi** [comment]
Yeah I saw! I will look at it when I have time, thanks a lot for the invitation 😁

**Comment by dwightjack** [comment]
Published as `vue-types@4.2.0-rc.1`

**Comment by dwightjack** [comment]
I totally forgot about this pending release. 

Now it is published in stable: https://github.com/dwightjack/vue-types/releases/tag/4.2.0
